plugins/kotlin/jvm-debugger/test/test/org/jetbrains/kotlin/idea/debugger/test/RemoteConnectionProvider.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.debugger.test
+
+import com.intellij.execution.configurations.JavaParameters
+import com.intellij.execution.configurations.RemoteConnection
+
+private const val PROVIDER_ENV = "INTELLIJ_REMOTE_CONNECTION_PROVIDER"
+private const val PROVIDER_PROPERTY = "intellij.remote.connection.provider"
+
+/**
+ * Provides a VM [RemoteConnection] for a [KotlinDescriptorTestCase]
+ */
+interface RemoteConnectionProvider {
+    /** Perform any setup operations */
+    fun setUp() {}
+
+    /** Perform any tear down operations */
+    fun tearDown() {}
+
+    /** Create a [RemoteConnection] object for a VM */
+    fun provideRemoteConnection(javaParameters: JavaParameters): RemoteConnection
+
+    companion object {
+        /**
+         * Returns a [RemoteConnectionProvider] if configured
+         *
+         * The return value is determined from the environment or a JVM property. The property can be a preset value (for example `art` or
+         * a class name that is expected to be in the classpath.
+         */
+        fun getInstance(testCase: KotlinDescriptorTestCase): RemoteConnectionProvider? {
+            return when (val provider = System.getProperty(PROVIDER_PROPERTY) ?: System.getenv(PROVIDER_ENV)) {
+                null -> null
+                "art" -> ArtRemoteConnectionProvider(testCase)
+                else -> loadProvider(provider, testCase)
+            }
+        }
+    }
+}
+
+private fun loadProvider(className: String, testCase: KotlinDescriptorTestCase): RemoteConnectionProvider {
```

Comment:
We are working on a way to run these tests from our own repository.

As we fix bugs, we expect to have to change the way the ART VM is launched. For example, we may want to add a certain command line option to the VM in order to activate some experimental feature/fix.

Since we can't anticipate what changes we will need, we would like to be able to have our own version of the `ArtAttacher` so we can iterate faster. When we stabilize it, we will merge it back upstream.